src/rotator_library/providers/antigravity_validators.py

```
@@ -0,0 +1,224 @@
+        
+        # Validate role
+        valid_roles = {'user', 'assistant', 'system'}
+        if message['role'] not in valid_roles:
+            raise ValueError(
+                f"Message at index {i} has invalid role '{message['role']}'. "
+                f"Must be one of: {', '.join(valid_roles)}"
+            )
+        
+        # Validate content
+        if not isinstance(message['content'], str):
+            raise TypeError(f"Message content must be a string, got {type(message['content']).__name__}")
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)